pkg/reconciler/v1alpha1/autoscaling/autoscaling.go

```diff
@@ -212,6 +213,20 @@ func (c *Reconciler) reconcile(ctx context.Context, key string, kpa *kpa.PodAuto
 	want := appliedScale
 	logger.Infof("KPA got=%v, want=%v", got, want)
 
+	var serviceLabel string
+	var configLabel string
+	if kpa.Labels != nil {
+		serviceLabel = kpa.Labels[serving.ServiceLabelKey]
+		configLabel = kpa.Labels[serving.ConfigurationLabelKey]
+	}
+	reporter, err := autoscaler.NewStatsReporter(kpa.Namespace, serviceLabel, configLabel, kpa.Name)
+	if err != nil {
+		return err
+	}
+
+	reporter.Report(autoscaler.ActualPodCountM, float64(got))
```

Comment:
Right, when I implemented this current version I started to wonder about the difference of `want` in this case and `status.replicas`.

My interpretation was as follows:

`DesiredPodCountM`--> What the autoscaler thinks we need. I actually would like to have this even pre-rate-limiting, so that this metric would reflect what the autoscaler ultimately thinks is desired.
`RequestedPodCountM` --> How many we request in the end, a.k.a `want` (after rate limiting, after applying scale bounds)
`ActualPodCountM` --> How many pods are actually there to serve, a.k.a. `got`

Hence the implementation as it is right now. Does that make sense? Could you elaborate on the difference of `spec.replicas` and `status.replicas` a bit? Is that basically the difference of setting the value and then waiting for a reconcile?

Reply:
Note: I have no skin in this game, just going off of what @josephburnett had said earlier :)

> Is that basically the difference of setting the value and then waiting for a reconcile?

I think more or less.

Reply:
I think that `status.replicas` is not so interesting.  We cover that with `got` calculated from the Endpoints.

The reason for the three values is to determine if the autoscaling signal is in our system (Knative Autoscaler) or Kubernetes.  And if it has been fulfilled.
1. in the Knative Autoscaler (`DesiredPodCountM` -- the output of `.Scale()`)
2. in the Pod scheduler (`RequestedPodCountM` -- the Deployment `spec.replicas`)
3. in the Cluster Autoscaler (`ActualPodCount` -- `got` from Endpoints)

E.g. suppose that we hit cluster resizing and then the master has to be resized (or whatever):
![image](https://user-images.githubusercontent.com/1103629/48085637-af37db80-e1af-11e8-8125-85d860e647fc.png)

Reply:
So I think the implementation as-is is correct.

/lgtm
/approve